sui/gmp.js

```diff
@@ -0,0 +1,136 @@
+const { getWallet, printWalletInfo, broadcast } = require('./sign-utils');
+
+async function sendCommand(keypair, client, contracts, args, options) {
+    const [destinationChain, destinationAddress, feeAmount, payload] = args;
+    const params = options.params;
+
+    const [testConfig, gasServiceConfig] = contracts;
+    const gasServicePackageId = gasServiceConfig.address;
+    const singletonObjectId = testConfig.objects.singleton;
+    const channelId = testConfig.objects.channelId;
+
+    const unitAmount = getUnitAmount(feeAmount);
+    const walletAddress = keypair.toSuiAddress();
+    const refundAddress = options.refundAddress || walletAddress;
+
+    const tx = new TransactionBlock();
+    const [coin] = tx.splitCoins(tx.gas, [unitAmount]);
+
+    tx.moveCall({
+        target: `${testConfig.address}::test::send_call`,
+        arguments: [
+            tx.object(singletonObjectId),
+            tx.pure(bcs.string().serialize(destinationChain).toBytes()),
+            tx.pure(bcs.string().serialize(destinationAddress).toBytes()),
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()),
+        ],
+    });
+
+    tx.moveCall({
+        target: `${gasServicePackageId}::gas_service::pay_gas`,
+        arguments: [
+            tx.object(gasServiceConfig.objects.GasService),
+            coin, // Coin<SUI>
+            tx.pure.address(channelId), // Channel address
+            tx.pure(bcs.string().serialize(destinationChain).toBytes()), // Destination chain
+            tx.pure(bcs.string().serialize(destinationAddress).toBytes()), // Destination address
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()), // Payload
+            tx.pure.address(refundAddress), // Refund address
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(params)).toBytes()), // Params
+        ],
+    });
+
+    const receipt = await broadcast(client, keypair, tx);
+
+    printInfo('Call sent', receipt.digest);
+}
+
+async function execute(keypair, client, contracts, args, options) {
+    const [testConfig, , axelarGatewayConfig] = contracts;
+
+    const [sourceChain, messageId, sourceAddress, payload] = args;
+
+    const singletonObjectId = testConfig.objects.singleton;
+    const gatewayObjectId = axelarGatewayConfig.objects.gateway;
+    const channelId = testConfig.objects.channelId;
+
+    const tx = new TransactionBlock();
+
+    const approvedMessage = tx.moveCall({
+        target: `${axelarGatewayConfig.address}::gateway::take_approved_message`,
+        arguments: [
+            tx.object(gatewayObjectId),
+            tx.pure(bcs.string().serialize(sourceChain).toBytes()),
+            tx.pure(bcs.string().serialize(messageId).toBytes()),
+            tx.pure(bcs.string().serialize(sourceAddress).toBytes()),
+            tx.pure.address(channelId),
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()),
+        ],
+    });
+
+    tx.moveCall({
```

Comment:
show an example of querying the Transaction from the discovery module to showcase how to execute generically